### PR TITLE
ICU-20469 Adding user guide for number skeletons.

### PR DESCRIPTION
**Pull request by sffc**
Also fixes a minor bug in the syntax and adds tests.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20469
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [x] Tests included
- [x] Documentation is changed or added

### CODE REVIEW

**Comment by sffc** [comment]
@pedberg-icu If you don't want this review, feel free to send it to someone else.  I put you as reviewer because you deal a lot with numbers but weren't quite as much involved in the skeleton syntax design.  There is a C API for this that maybe you can start using more. :)

File: docs/userguide/format_parse/numbers/skeletons.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+## Examples
+
+| Skeleton | Input | en-US Output | Comments |
+|---|---|---|---|
+| `percent` | 25 | 25% |
+| `.00` | 25 | 25.00 | Equivalent to Precision::fixedFraction(2) |
+| `percent .00` | 25 | 25.00% |
+| `scale/100` | 0.3 | 30 | Multiply by 100 before formatting |
+| `percent scale/100` | 0.3 | 30% |
+| `measure-unit/length-meter` | 5 | 5 m |
```
Comment (pedberg-icu) [inline docs/userguide/format_parse/numbers/skeletons.md]: Why is the output here not "5 meters"? Is there a measure-unit width default to short or narrow forms (if so, that should be made clear somewhere)?

Comment (sffc) [reply docs/userguide/format_parse/numbers/skeletons.md]: Yes, unit width is short by default for all unit types (currency, percent, and measure unit).  Added a note about it.

**Comment by sffc** [comment]
Squashed; please approve

**Pull request merged**